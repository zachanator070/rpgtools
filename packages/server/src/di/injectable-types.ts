import {DatabaseContext} from "../dal/database-context";

export const INJECTABLE_TYPES = {

	DomainEntity: Symbol.for("DomainEntity"),

	// domain entities
	Article: Symbol.for("Article"),
	Chunk: Symbol.for("Chunk"),
	File: Symbol.for("File"),
	Game: Symbol.for("Game"),
	Image: Symbol.for("Image"),
	Item: Symbol.for("Item"),
	Model: Symbol.for("Model"),
	Monster: Symbol.for("Monster"),
	PermissionAssignment: Symbol.for("PermissionAssignment"),
	Person: Symbol.for("Person"),
	Pin: Symbol.for("Pin"),
	Place: Symbol.for("Place"),
	Role: Symbol.for("Role"),
	ServerConfig: Symbol.for("ServerConfig"),
	User: Symbol.for("User"),
	WikiFolder: Symbol.for("WikiFolder"),
	World: Symbol.for("World"),
	Event: Symbol.for("Event"),
	Calendar: Symbol.for("Calendar"),

	// entity factories
	ArticleFactory: Symbol.for("ArticleFactory"),
	AclFactory: Symbol.for('AclFactory'),
	ChunkFactory: Symbol.for("ChunkFactory"),
	FileFactory: Symbol.for("FileFactory"),
	GameFactory: Symbol.for("GameFactory"),
	ImageFactory: Symbol.for("ImageFactory"),
	ItemFactory: Symbol.for("ItemFactory"),
	ModelFactory: Symbol.for("ModelFactory"),
	MonsterFactory: Symbol.for("MonsterFactory"),
	PersonFactory: Symbol.for("PersonFactory"),
	PinFactory: Symbol.for("PinFactory"),
	PlaceFactory: Symbol.for("PlaceFactory"),
	RoleFactory: Symbol.for("RoleFactory"),
	ServerConfigFactory: Symbol.for("ServerConfigFactory"),
	UserFactory: Symbol.for("UserFactory"),
	WikiFolderFactory: Symbol.for("WikiFolderFactory"),
	WikiPageFactory: Symbol.for("WikiPageFactory"),
	WorldFactory: Symbol.for("WorldFactory"),
	EventWikiFactory: Symbol.for("EventFactory"),
	CalendarFactory: Symbol.for("CalendarFactory"),

	CharacterFactory: Symbol.for('CharacterFactory'),
	CharacterAttributeFactory: Symbol.for('CharacterAttributeFactory'),
	PathNodeFactory: Symbol.for('NodePathFactory'),
	StrokeFactory: Symbol.for('StrokeFactory'),
	FogStrokeFactory: Symbol.for('FogStrokeFactory'),
	MessageFactory: Symbol.for('MessageFactory'),
	InGameModelFactory: Symbol.for('InGameModelFactory'),

	AgeFactory: Symbol.for("AgeFactory"),
	MonthFactory: Symbol.for("MonthFactory"),
	DayOfTheWeekFactory: Symbol.for("DayOfTheWeekFactory"),

	// repositories
	ArticleRepository: Symbol.for("ArticleRepository"),
	ChunkRepository: Symbol.for("ChunkRepository"),
	FileRepository: Symbol.for("FileRepository"),
	GameRepository: Symbol.for("GameRepository"),
	ImageRepository: Symbol.for("ImageRepository"),
	ItemRepository: Symbol.for("ItemRepository"),
	ModelRepository: Symbol.for("ModelRepository"),
	MonsterRepository: Symbol.for("MonsterRepository"),
	PersonRepository: Symbol.for("PersonRepository"),
	PinRepository: Symbol.for("PinRepository"),
	PlaceRepository: Symbol.for("PlaceRepository"),
	RoleRepository: Symbol.for("RoleRepository"),
	ServerConfigRepository: Symbol.for("ServerConfigRepository"),
	UserRepository: Symbol.for("UserRepository"),
	WikiFolderRepository: Symbol.for("WikiFolderRepository"),
	WikiPageRepository: Symbol.for("WikiPageRepository"),
	WorldRepository: Symbol.for("WorldRepository"),
	EventWikiRepository: Symbol.for("EventWikiRepository"),
	CalendarRepository: Symbol.for("CalendarRepository"),
	StrokeRepository: Symbol.for('StrokeRepository'),
	FogStrokeRepository: Symbol.for('FogStrokeRepository'),

	SqlPermissionControlledRepository: Symbol.for('SqlPermissionControlledRepository'),

	// authorization policies
	ArticleAuthorizationPolicy: Symbol.for("ArticleAuthorizationPolicy"),
	ChunkAuthorizationPolicy: Symbol.for("ChunkAuthorizationPolicy"),
	FileAuthorizationPolicy: Symbol.for("FileAuthorizationPolicy"),
	GameAuthorizationPolicy: Symbol.for("GameAuthorizationPolicy"),
	ImageAuthorizationPolicy: Symbol.for("ImageAuthorizationPolicy"),
	ItemAuthorizationPolicy: Symbol.for("ItemAuthorizationPolicy"),
	ModelAuthorizationPolicy: Symbol.for("ModelAuthorizationPolicy"),
	MonsterAuthorizationPolicy: Symbol.for("MonsterAuthorizationPolicy"),
	PersonAuthorizationPolicy: Symbol.for("PersonAuthorizationPolicy"),
	PinAuthorizationPolicy: Symbol.for("PinAuthorizationPolicy"),
	PlaceAuthorizationPolicy: Symbol.for("PlaceAuthorizationPolicy"),
	RoleAuthorizationPolicy: Symbol.for("RoleAuthorizationPolicy"),
	ServerConfigAuthorizationPolicy: Symbol.for("ServerConfigAuthorizationPolicy"),
	UserAuthorizationPolicy: Symbol.for("UserAuthorizationPolicy"),
	WikiFolderAuthorizationPolicy: Symbol.for("WikiFolderAuthorizationPolicy"),
	WikiPageAuthorizationPolicy: Symbol.for("WikiPageAuthorizationPolicy"),
	WorldAuthorizationPolicy: Symbol.for("WorldAuthorizationPolicy"),
	CalendarAuthorizationPolicy: Symbol.for("CalendarAuthorizationPolicy"),
	FogStrokeAuthorizationPolicy: Symbol.for('FogStrokeAuthorizationPolicy'),
	StrokeAuthorizationPolicy: Symbol.for('StrokeAuthorizationPolicy'),

	// archive repositories
	ArchiveArticleRepository: Symbol.for("ArchiveArticleRepository"),
	ArchiveChunkRepository: Symbol.for("ArchiveChunkRepository"),
	ArchiveGameRepository: Symbol.for("ArchiveGameRepository"),
	ArchiveImageRepository: Symbol.for("ArchiveImageRepository"),
	ArchiveItemRepository: Symbol.for("ArchiveItemRepository"),
	ArchiveModelRepository: Symbol.for("ArchiveModelRepository"),
	ArchiveMonsterRepository: Symbol.for("ArchiveMonsterRepository"),
	ArchivePermissionAssignmentRepository: Symbol.for("ArchivePermissionAssignmentRepository"),
	ArchivePersonRepository: Symbol.for("ArchivePersonRepository"),
	ArchivePinRepository: Symbol.for("ArchivePinRepository"),
	ArchivePlaceRepository: Symbol.for("ArchivePlaceRepository"),
	ArchiveRoleRepository: Symbol.for("ArchiveRoleRepository"),
	ArchiveServerConfigRepository: Symbol.for("ArchiveServerConfigRepository"),
	ArchiveUserRepository: Symbol.for("ArchiveUserRepository"),
	ArchiveWikiFolderRepository: Symbol.for("ArchiveWikiFolderRepository"),
	ArchiveWikiPageRepository: Symbol.for("ArchiveWikiPageRepository"),
	ArchiveWorldRepository: Symbol.for("ArchiveWorldRepository"),
	ArchiveFileRepository: Symbol.for("ArchiveFileRepository"),

	// services
	AuthenticationService: Symbol.for("AuthenticationService"),
	AuthorizationService: Symbol.for("AuthorizationService"),
	ContentExportService: Symbol.for("ContentExportService"),
	ContentImportService: Symbol.for("ContentImportService"),
	GameService: Symbol.for("GameService"),
	ImageService: Symbol.for("ImageService"),
	ModelService: Symbol.for("ModelService"),
	RoleService: Symbol.for("RoleService"),
	ServerConfigService: Symbol.for("ServerConfigService"),
	SrdImportService: Symbol.for("SrdImportService"),
	UserService: Symbol.for("UserService"),
	WikiFolderService: Symbol.for("WikiFolderService"),
	WikiPageService: Symbol.for("WikiPageService"),
	WorldService: Symbol.for("WorldService"),

	// session contexts
	SessionContextFactory: Symbol.for("SessionContextFactory"),
	SecurityContextFactory: Symbol.for("SecurityContextFactory"),

	// 5e
	Open5eApiClient: Symbol.for("Open5eApiClient"),
	Dnd5eApiClient: Symbol.for("Dnd5eApiClient"),

	// delta
	DeltaFactory: Symbol.for("DeltaFactory"),

	// cache
	Cache: Symbol.for("Cache"),

	// archive
	ArchiveFactory: Symbol.for("ArchiveFactory"),

	ZipArchiveFactory: Symbol.for("ZipArchiveFactory"),
	ZipArchive: Symbol.for('ZipArchive'),

	// event publisher
	EventPublisher: Symbol.for("EventPublisher"),

	// server
	ApiServer: Symbol.for("ApiServer"),
	ServerProperties: Symbol.for("ServerProperties"),
	RpgToolsServer: Symbol.for('RpgToolsServer'),
	DbEngine: Symbol.for("DbEngine"),
	SqlDbEngine: Symbol.for('SqlDbEngine'),

	// data loaders
	ArticleDataLoader: Symbol.for("ArticleDataLoader"),
	ChunkDataLoader: Symbol.for("ChunkDataLoader"),
	FileDataLoader: Symbol.for("FileDataLoader"),
	GameDataLoader: Symbol.for("GameDataLoader"),
	ImageDataLoader: Symbol.for("ImageDataLoader"),
	ItemDataLoader: Symbol.for("ItemDataLoader"),
	ModelDataLoader: Symbol.for("ModelDataLoader"),
	MonsterDataLoader: Symbol.for("MonsterDataLoader"),
	PermissionAssignmentDataLoader: Symbol.for("PermissionAssignmentDataLoader"),
	PersonDataLoader: Symbol.for("PersonDataLoader"),
	PinDataLoader: Symbol.for("PinDataLoader"),
	PlaceDataLoader: Symbol.for("PlaceDataLoader"),
	RoleDataLoader: Symbol.for("RoleDataLoader"),
	ServerConfigDataLoader: Symbol.for("ServerConfigDataLoader"),
	UserDataLoader: Symbol.for("UserDataLoader"),
	WikiFolderDataLoader: Symbol.for("WikiFolderDataLoader"),
	WikiPageDataLoader: Symbol.for("WikiPageDataLoader"),
	WorldDataLoader: Symbol.for("WorldDataLoader"),
	CalendarDataLoader: Symbol.for('CalendarDataLoader'),
	EventDataLoader: Symbol.for('EventDataLoader'),

	// seeders
	RoleSeeder: Symbol.for("RoleSeeder"),
	ServerConfigSeeder: Symbol.for("ServerConfigSeeder"),

	// database context
	DatabaseContext: Symbol.for("DatabaseContext"),
	DatabaseContextFactory: Symbol.for("DatabaseContextFactory"),

	// repo mapper
	EntityMapper: Symbol.for("EntityMapper"),

	// mongodb
	FilterFactory: Symbol.for("FilterFactory"),
	MongoDbMigrationV40: Symbol.for("MongoDbMigrationV40"),


};
