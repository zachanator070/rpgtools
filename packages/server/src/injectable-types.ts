export const INJECTABLE_TYPES = {
	ArticleRepository: Symbol.for("ArticleRepository"),
	ChunkRepository: Symbol.for("ChunkRepository"),
	GameRepository: Symbol.for("GameRepository"),
	ImageRepository: Symbol.for("ImageRepository"),
	ItemRepository: Symbol.for("ItemRepository"),
	ModelRepository: Symbol.for("ModelRepository"),
	MonsterRepository: Symbol.for("MonsterRepository"),
	PermissionAssignmentRepository: Symbol.for("PermissionAssignmentRepository"),
	PersonRepository: Symbol.for("PersonRepository"),
	PinRepository: Symbol.for("PinRepository"),
	PlaceRepository: Symbol.for("PlaceRepository"),
	RoleRepository: Symbol.for("RoleRepository"),
	ServerConfigRepository: Symbol.for("ServerConfigRepository"),
	UserRepository: Symbol.for("UserRepository"),
	WikiFolderRepository: Symbol.for("WikiFolderRepository"),
	WikiPageRepository: Symbol.for("WikiPageRepository"),
	WorldRepository: Symbol.for("WorldRepository"),
	FileRepository: Symbol.for("FileRepository"),

	AuthenticationService: Symbol.for("AuthenticationService"),
	AuthorizationService: Symbol.for("AuthorizationService"),
	ContentExportService: Symbol.for("ContentExportService"),
	ContentImportService: Symbol.for("ContentImportService"),
	GameService: Symbol.for("GameService"),
	ImageService: Symbol.for("ImageService"),
	ModelService: Symbol.for("ModelService"),
	ServerConfigService: Symbol.for("ServerConfigService"),
	SrdImportService: Symbol.for("SrdImportService"),
	UserService: Symbol.for("UserService"),
	WikiFolderService: Symbol.for("WikiFolderService"),
	WikiPageService: Symbol.for("WikiPageService"),
	WorldService: Symbol.for("WorldService"),

	SessionContextFactory: Symbol.for("SessionContextFactory"),
	Open5eApiClient: Symbol.for("Open5eApiClient"),

	Cache: Symbol.for("Cache"),
};
