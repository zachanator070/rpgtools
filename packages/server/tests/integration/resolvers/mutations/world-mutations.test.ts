import {DefaultTestingContext} from "../../default-testing-context";
import {
	CREATE_PIN,
	CREATE_WORLD, DELETE_CALENDAR,
	DELETE_PIN,
	RENAME_WORLD,
	UPDATE_PIN,
	UPSERT_CALENDAR
} from "@rpgtools/common/src/gql-mutations";
import {container} from "../../../../src/di/inversify";
import {TEST_INJECTABLE_TYPES} from "../../injectable-types";

process.env.TEST_SUITE = "world-mutations-test";

describe("world-mutations", () => {
	describe("with world and logged in", () => {
		const testingContext = container.get<DefaultTestingContext>(TEST_INJECTABLE_TYPES.DefaultTestingContext);

		beforeEach(async () => {
			await testingContext.reset();
		});

		test("create world", async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: CREATE_WORLD,
				variables: { name: "Earth", public: false },
			});
			expect(result).toMatchSnapshot({
				data: {
					createWorld: {
						_id: expect.any(String),
						wikiPage: {
							_id: expect.any(String),
						},
					},
				},
				errors: undefined,
			});
		});

		test("rename world", async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: RENAME_WORLD,
				variables: { worldId: testingContext.world._id, newName: "Azeroth" },
			});
			expect(result).toMatchSnapshot({
				data: {
					renameWorld: {
						_id: expect.any(String),
					},
				},
				errors: undefined,
			});
		});

		test("create pin", async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: CREATE_PIN,
				variables: {
					mapId: testingContext.world.wikiPage,
					x: 0,
					y: 0,
					wikiId: testingContext.world.wikiPage,
				},
			});
			expect(result).toMatchSnapshot({
				data: {
					createPin: {
						_id: expect.any(String),
						page: {
							_id: expect.any(String),
						},
						map: {
							_id: expect.any(String),
						}
					},
				},
				errors: undefined,
			});
		});

		test("update pin", async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: UPDATE_PIN,
				variables: { pinId: testingContext.pin._id, pageId: testingContext.otherPage._id },
			});
			expect(result).toMatchSnapshot({
				data: {
					updatePin: {
						_id: expect.any(String),
						page: {
							_id: expect.any(String),
						},
						map: {
							_id: expect.any(String),
						}
					},
				},
				errors: undefined,
			});
		});

		test("delete pin", async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: DELETE_PIN,
				variables: { pinId: testingContext.pin._id },
			});
			expect(result).toMatchSnapshot({
				data: {
					deletePin: {
						_id: expect.any(String),
						page: {
							_id: expect.any(String),
						},
						map: {
							_id: expect.any(String),
						}
					},
				},
				errors: undefined,
			});
		});

		test("create calendar", async () => {
			const calendarName = 'new calendar';
			const ageName = 'new age';
			const monthName = 'new month';
			const dayName = 'new day';
			const result = await testingContext.server.executeGraphQLQuery({
				query: UPSERT_CALENDAR,
				variables: {
					name: calendarName,
					world: testingContext.world._id,
					ages: [{
						name: ageName,
						numYears: 1000,
						months: [{
							name: monthName,
							numDays: 30
						}],
						daysOfTheWeek: [{
							name: dayName
						}]
					}]
				}
			});
			expect(result).toMatchSnapshot({
				data: {
					upsertCalendar: {
						_id: expect.any(String),
						name: calendarName,
						ages: [{
							_id: expect.any(String),
							name: ageName,
							numYears: 1000,
							months: [{
								_id: expect.any(String),
								name: monthName,
								numDays: 30
							}],
							daysOfTheWeek: [{
								_id: expect.any(String),
								name: dayName
							}]
						}],
						accessControlList: expect.arrayContaining([
							{
								permission: expect.any(String),
								principal: {
									_id: expect.any(String),
									name: expect.any(String)
								},
								principalType: expect.any(String)
							}
						])
					},
				},
				errors: undefined,
			});
		});

		test("update calendar age years", async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: UPSERT_CALENDAR,
				variables: {
					name: testingContext.calendar1.name,
					calendarId: testingContext.calendar1._id,
					world: testingContext.world._id,
					ages: [{
						name: testingContext.calendar1.ages[0].name,
						_id: testingContext.calendar1.ages[0]._id,
						numYears: 1000,
						months: [{
							name: testingContext.calendar1.ages[0].months[0].name,
							_id: testingContext.calendar1.ages[0].months[0]._id,
							numDays: 60
						}],
						daysOfTheWeek: [{
							name: testingContext.calendar1.ages[0].daysOfTheWeek[0].name,
							_id: testingContext.calendar1.ages[0].daysOfTheWeek[0]._id
						}]
					}]
				}
			});
			expect(result).toMatchSnapshot({
				data: {
					upsertCalendar: {
						_id: expect.any(String),
						name: testingContext.calendar1.name,
						ages: [{
							_id: expect.any(String),
							name: testingContext.calendar1.ages[0].name,
							numYears: 1000,
							months: [{
								name: testingContext.calendar1.ages[0].months[0].name,
								_id: expect.any(String),
								numDays: 60
							}],
							daysOfTheWeek: [{
								_id: expect.any(String),
								name: testingContext.calendar1.ages[0].daysOfTheWeek[0].name
							}]
						}],
						accessControlList: expect.arrayContaining([
							{
								permission: expect.any(String),
								principal: {
									_id: expect.any(String),
									name: expect.any(String)
								},
								principalType: expect.any(String)
							}
						])
					},
				},
				errors: undefined,
			});
		});

		test("update calendar delete month", async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: UPSERT_CALENDAR,
				variables: {
					name: testingContext.calendar1.name,
					calendarId: testingContext.calendar1._id,
					world: testingContext.world._id,
					ages: [{
						name: testingContext.calendar1.ages[0].name,
						_id: testingContext.calendar1.ages[0]._id,
						numYears: 100,
						months: [],
						daysOfTheWeek: [{
							name: testingContext.calendar1.ages[0].daysOfTheWeek[0].name,
							_id: testingContext.calendar1.ages[0].daysOfTheWeek[0]._id
						}]
					}]
				}
			});
			expect(result).toMatchSnapshot({
				data: {
					upsertCalendar: {
						_id: expect.any(String),
						name: testingContext.calendar1.name,
						ages: [{
							_id: expect.any(String),
							name: testingContext.calendar1.ages[0].name,
							numYears: 100,
							months: [],
							daysOfTheWeek: [{
								_id: expect.any(String),
								name: testingContext.calendar1.ages[0].daysOfTheWeek[0].name
							}]
						}],
						accessControlList: expect.arrayContaining([
							{
								permission: expect.any(String),
								principal: {
									_id: expect.any(String),
									name: expect.any(String)
								},
								principalType: expect.any(String)
							}
						])
					},
				},
				errors: undefined,
			});
		});

		test('delete calendar', async () => {
			const result = await testingContext.server.executeGraphQLQuery({
				query: DELETE_CALENDAR,
				variables: {
					calendarId: testingContext.calendar1._id,
				}
			});
			expect(result).toMatchSnapshot({
				data: {
					deleteCalendar: {
						_id: expect.any(String),
					},
				},
				errors: undefined,
			});
		});

		describe("not logged in", () => {
			beforeEach(() => {
				testingContext.mockSessionContextFactory.useAnonUser();
			});
			test("create world no permissions", async () => {
				const result = await testingContext.server.executeGraphQLQuery({
					query: CREATE_WORLD,
					variables: { name: "Earth", public: false },
				});
				expect(result).toMatchSnapshot({
					errors: expect.arrayContaining([expect.any(Object)]),
				});
			});

			test("rename world no permission", async () => {
				const result = await testingContext.server.executeGraphQLQuery({
					query: RENAME_WORLD,
					variables: { worldId: testingContext.world._id, newName: "Azeroth" },
				});
				expect(result).toMatchSnapshot({
					errors: expect.arrayContaining([expect.any(Object)]),
				});
			});

			test("create pin no permission", async () => {
				const result = await testingContext.server.executeGraphQLQuery({
					query: CREATE_PIN,
					variables: {
						mapId: testingContext.world.wikiPage,
						x: 0,
						y: 0,
						wikiId: testingContext.world.wikiPage,
					},
				});
				expect(result).toMatchSnapshot({
					errors: expect.arrayContaining([expect.any(Object)]),
				});
			});

			test("update pin no permission", async () => {
				const result = await testingContext.server.executeGraphQLQuery({
					query: UPDATE_PIN,
					variables: { pinId: testingContext.pin._id, pageId: testingContext.otherPage._id },
				});
				expect(result).toMatchSnapshot({
					errors: expect.arrayContaining([expect.any(Object)]),
				});
			});

			test("delete pin no permission", async () => {
				const result = await testingContext.server.executeGraphQLQuery({
					query: DELETE_PIN,
					variables: { pinId: testingContext.pin._id },
				});
				expect(result).toMatchSnapshot({
					errors: expect.arrayContaining([expect.any(Object)]),
				});
			});

			test("update calendar no permission", async () => {
				const result = await testingContext.server.executeGraphQLQuery({
					query: UPSERT_CALENDAR,
					variables: {
						name: testingContext.calendar1.name,
						calendarId: testingContext.calendar1._id,
						world: testingContext.world._id,
						ages: []
					}
				});
				expect(result).toMatchSnapshot({
					errors: expect.arrayContaining([expect.any(Object)]),
				});
			});

			test("create calendar no permission", async () => {
				const result = await testingContext.server.executeGraphQLQuery({
					query: UPSERT_CALENDAR,
					variables: {
						name: testingContext.calendar1.name,
						world: testingContext.world._id,
						ages: []
					}
				});
				expect(result).toMatchSnapshot({
					errors: expect.arrayContaining([expect.any(Object)]),
				});
			});
		});
	});
});
