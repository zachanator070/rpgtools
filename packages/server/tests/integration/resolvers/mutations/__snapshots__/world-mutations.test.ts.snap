// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`world-mutations with world and logged in create calendar 1`] = `
{
  "data": {
    "upsertCalendar": {
      "_id": Any<String>,
      "accessControlList": ArrayContaining [
        {
          "permission": Any<String>,
          "principal": {
            "_id": Any<String>,
            "name": Any<String>,
          },
          "principalType": Any<String>,
        },
      ],
      "ages": [
        {
          "_id": Any<String>,
          "daysOfTheWeek": [
            {
              "_id": Any<String>,
              "name": "new day",
            },
          ],
          "months": [
            {
              "_id": Any<String>,
              "name": "new month",
              "numDays": 30,
            },
          ],
          "name": "new age",
          "numYears": 1000,
        },
      ],
      "canAdmin": true,
      "canWrite": true,
      "name": "new calendar",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in create pin 1`] = `
{
  "data": {
    "createPin": {
      "_id": Any<String>,
      "canWrite": true,
      "map": {
        "_id": Any<String>,
        "name": "Earth",
      },
      "page": {
        "_id": Any<String>,
        "name": "Earth",
        "type": "Place",
      },
      "x": 0,
      "y": 0,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in create world 1`] = `
{
  "data": {
    "createWorld": {
      "_id": Any<String>,
      "wikiPage": {
        "_id": Any<String>,
      },
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in delete calendar 1`] = `
{
  "data": {
    "deleteCalendar": {
      "_id": Any<String>,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in delete pin 1`] = `
{
  "data": {
    "deletePin": {
      "_id": Any<String>,
      "canWrite": true,
      "map": {
        "_id": Any<String>,
        "name": "Earth",
      },
      "page": {
        "_id": Any<String>,
        "name": "other page",
        "type": "Article",
      },
      "x": 0,
      "y": 0,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in not logged in create calendar no permission 1`] = `
{
  "data": null,
  "errors": ArrayContaining [
    Any<Object>,
  ],
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in not logged in create pin no permission 1`] = `
{
  "data": null,
  "errors": ArrayContaining [
    Any<Object>,
  ],
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in not logged in create world no permissions 1`] = `
{
  "data": null,
  "errors": ArrayContaining [
    Any<Object>,
  ],
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in not logged in delete pin no permission 1`] = `
{
  "data": null,
  "errors": ArrayContaining [
    Any<Object>,
  ],
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in not logged in rename world no permission 1`] = `
{
  "data": null,
  "errors": ArrayContaining [
    Any<Object>,
  ],
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in not logged in update calendar no permission 1`] = `
{
  "data": null,
  "errors": ArrayContaining [
    Any<Object>,
  ],
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in not logged in update pin no permission 1`] = `
{
  "data": null,
  "errors": ArrayContaining [
    Any<Object>,
  ],
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in rename world 1`] = `
{
  "data": {
    "renameWorld": {
      "_id": Any<String>,
      "name": "Azeroth",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in update calendar age years 1`] = `
{
  "data": {
    "upsertCalendar": {
      "_id": Any<String>,
      "accessControlList": ArrayContaining [
        {
          "permission": Any<String>,
          "principal": {
            "_id": Any<String>,
            "name": Any<String>,
          },
          "principalType": Any<String>,
        },
      ],
      "ages": [
        {
          "_id": Any<String>,
          "daysOfTheWeek": [
            {
              "_id": Any<String>,
              "name": "Monday",
            },
          ],
          "months": [
            {
              "_id": Any<String>,
              "name": "month 1",
              "numDays": 60,
            },
          ],
          "name": "age 1",
          "numYears": 1000,
        },
      ],
      "canAdmin": true,
      "canWrite": true,
      "name": "Earth calendar",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in update calendar delete month 1`] = `
{
  "data": {
    "upsertCalendar": {
      "_id": Any<String>,
      "accessControlList": ArrayContaining [
        {
          "permission": Any<String>,
          "principal": {
            "_id": Any<String>,
            "name": Any<String>,
          },
          "principalType": Any<String>,
        },
      ],
      "ages": [
        {
          "_id": Any<String>,
          "daysOfTheWeek": [
            {
              "_id": Any<String>,
              "name": "Monday",
            },
          ],
          "months": [],
          "name": "age 1",
          "numYears": 100,
        },
      ],
      "canAdmin": true,
      "canWrite": true,
      "name": "Earth calendar",
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;

exports[`world-mutations with world and logged in update pin 1`] = `
{
  "data": {
    "updatePin": {
      "_id": Any<String>,
      "canWrite": true,
      "map": {
        "_id": Any<String>,
        "name": "Earth",
      },
      "page": {
        "_id": Any<String>,
        "name": "other page",
        "type": "Article",
      },
      "x": 0,
      "y": 0,
    },
  },
  "errors": undefined,
  "extensions": undefined,
  "http": {
    "headers": Headers {
      Symbol(map): {},
    },
  },
}
`;
